### PR TITLE
Ruby enumerable methods

### PR DESCRIPTION
**Pull request by pbkabali**
## What this PR does

- Adds the reconstructed methods from the ruby enumerable module

## Screenshot

image

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

Great work on this.  You did well for the first review on this. 

There are just two issues wrong in the project. Kindly refer to the below-highlighted items.👇

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    my_each { |i| result.push(yield(i)) }
+    result
+  end
+
+  #------------ my_inject -----------------------------
+
+  def my_inject(*arg)
+    validator = false
+    counter = Array(self)[0]
+    if (arg[0].class == Symbol) || arg[0].nil?
```
Comment (mikenath223) [inline enumerables.rb]: - [ ] my_inject is expected to raise a "LocalJumpError" when no block or argument is given. Please fix it.

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      end
+    end
+    counter
+  end
+end
+# rubocop:enable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+
+def my_multiply_els(arr)
+  arr.my_inject(:*)
+end
```
Comment (mikenath223) [inline enumerables.rb]: - [ ] Please rename this to multiply_els.

Comment (diazgio) [reply enumerables.rb]: I already did this two changes required thanks for your help :+1:

**Comment by pbkabali** [comment]
Hi @mikenath223, thank you for your review. We have worked on the feedback and submitted another review request.

**Review by christianotieno — APPROVED** [review]
# PROJECT APPROVED :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
***All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.***
- Congratulations on the splendid job :+1:
- You can now merge these changes.
- I wish you all the very best in your next tasks 🥇 
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)

**Pull request merged**